compiler/codegen/OMRCodeGenerator.cpp

```diff
@@ -1275,6 +1275,16 @@ OMR::CodeGenerator::removeUnusedLocals()
    }
 
 
+void
+OMR::CodeGenerator::expandInstructions()
+   {
+   for (TR::Instruction *instr = self()->getFirstInstruction(); instr = instr->getNext(); instr)
```

Comment:
I think you have the condition and iteration expressions reversed.

Reply:
Um, on second thought, what you have might work, though it's an atypical way of writing it.

Reply:
Oops, you're right, I somehow reversed these and didn't notice :sweat_smile:. Also, as you noticed, this does _almost_ work correctly, which explains how I missed it when testing, but it will actually miss calling `expandInstruction` on the very first instruction so I still need to fix this.

Reply:
Fixed by a1398596c7e866a7e44a02dcaed632d9276a35e7.